hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -282,8 +282,12 @@ public String toString() {
     public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
         ClientStatistics clientStatistics = statsRef.get();
         if (clientStatistics != null && clientStatistics.metricsBlob() != null) {
-            long timestamp = clientStatistics.timestamp();
             byte[] metricsBlob = clientStatistics.metricsBlob();
+            if (metricsBlob.length == 0) {
```

Comment:
Alternatively, we can add this to the surrounding if.

Reply:
Yup, I didn't do it as it already includes two checks